pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -182,26 +183,26 @@ func (a *acrProvider) Enabled() bool {
 }
 
 func (a *acrProvider) Provide(image string) credentialprovider.DockerConfig {
+	klog.V(2).Infof("try to provide secret for image %s", image)
 	cfg := credentialprovider.DockerConfig{}
-	ctx, cancel := getContextWithCancel()
-	defer cancel()
 
 	if a.config.UseManagedIdentityExtension {
-		klog.V(4).Infof("listing registries")
-		result, err := a.registryClient.List(ctx)
+		// Parse login server from parameter `image`. An example of parameter `image`: foo.azurecr.io/bar/imageName:version
+		regEx, err := regexp.Compile(`.*\.azurecr\.io|.*\.azurecr\.cn|.*\.azurecr\.de|.*\.azurecr\.us`)
 		if err != nil {
-			klog.Errorf("Failed to list registries: %v", err)
-			return cfg
+			klog.Error("Unexpected error: failed to construct regular expression: %v", err)
```

Comment:
only log error here?

Reply:
Please refer to latest commit.

Reply:
It would be OK to only log error, the error is not fatal and we can't do anything about it. For the worst case, `cfg` is not correctly filled and pulling image will fail, which is acceptable.